File: PHP/03_storage_json.php
<?php


echo "--------------------------------------------------------------------------\n";
echo "\tCouchbase JSON Doc Storage\n";
echo "--------------------------------------------------------------------------\n";


$cb = new Couchbase();


$mydoc =  array("doctype" => "test", "name" => "John Smith");

echo "Original Document:\n";
var_dump($mydoc);

echo "\nSet Document (Hash -> JSON)\n";
// store a json doc (encode it)
$cb->set("mydoc", json_encode($mydoc));


echo "Retrieve Document (JSON -> Hash), also set a 10 second TTL\n";
// retrieve and decode json doc
$doc = $cb->getAndTouch("mydoc",10);

echo "\n--------------\n";
echo "JSON String: " . $doc;
echo "\n--------------\n";
var_dump(json_decode( $doc,true)); // (as array, not object)

echo "--------------------------------------------------------------------------\n";

?>

File: Ruby/03_storage_json.rb
#!/usr/bin/env ruby

require "rubygems"
require "couchbase"
require "json"

require "rainbow"
require "pp"


system 'clear'
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts "Couchbase JSON Doc Storage".bright.foreground(:red)
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts

# establish connection
cb = Couchbase.connect

mydoc = {
	 :doctype => "test",
   :name => "John Smith"
}

puts "Set Document (Hash -> JSON)".bright
puts

# store a json doc (encode it)
cb.set("mydoc", mydoc)

puts "Retrieve Document (JSON -> Hash), also set a 10 second TTL".bright
puts

# retrieve and decode json doc
doc = cb.get("mydoc", :ttl => 10)
pp doc
puts "doc[\"name\"] = " + doc["name"]

puts
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts